python/ray/train/base_trainer.py

```diff
@@ -426,11 +419,11 @@ def _trainable_func(self, config, reporter, checkpoint_dir):
             @classmethod
             def default_resource_request(cls, config):
                 updated_scaling_config = config.get("scaling_config", scaling_config)
-                scaling_config_dataclass = (
-                    trainer_cls._validate_and_get_scaling_config_data_class(
-                        updated_scaling_config
-                    )
+                if isinstance(updated_scaling_config, dict):
+                    updated_scaling_config = ScalingConfig(**updated_scaling_config)
```

Comment:
When is this expected to be a dict vs dataclass?

Reply:
as of now, dataclass when `scaling_config` is passed directly to trainer and dict when `scaling_config` is passed through `param_space`

Reply:
Ah got it, just noticed since this is a class method it needs to duplicate the logic in lines 372-374. 

Could you add a quick comment in both places to document this behavior?